File: frontend/src/types/conversation.ts
// Conversation and message types

export interface Conversation {
  id: string;
  agent_id: string;
  title: string;
  created_at: string;
  updated_at: string;
  message_count: number;
}

export interface ConversationCreate {
  agent_id: string;
  title?: string;
}

export interface Message {
  id: string;
  conversation_id: string;
  role: 'user' | 'assistant' | 'system';
  content: string;
  metadata?: Record<string, any>;
  created_at: string;
}

export interface ChatRequest {
  message: string;
}

export interface ChatResponse {
  user_message: Message;
  assistant_message: Message;
}

File: backend/schemas/conversation.py
"""Pydantic schemas for conversation and message API"""

from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field


# ============================================================================
# Conversation Schemas
# ============================================================================

class ConversationCreate(BaseModel):
    agent_id: str
    title: Optional[str] = None


class ConversationUpdate(BaseModel):
    title: Optional[str] = None


class ConversationResponse(BaseModel):
    id: str
    agent_id: str
    title: Optional[str]
    created_at: datetime
    updated_at: datetime
    message_count: int


# ============================================================================
# Message Schemas
# ============================================================================

class MessageCreate(BaseModel):
    content: str
    role: str = "user"  # 'user', 'assistant', 'system'


class MessageResponse(BaseModel):
    id: str
    conversation_id: str
    role: str
    content: str
    metadata: Optional[dict] = None
    created_at: datetime


class ChatRequest(BaseModel):
    """Request to send a message and get LLM response"""
    message: str


class ChatResponse(BaseModel):
    """Response containing both user message and assistant reply"""
    user_message: MessageResponse
    assistant_message: MessageResponse
